tools/nnyeah/nnyeah/Reworker.cs

```diff
@@ -53,59 +53,60 @@ public static bool NeedsReworking (ModuleDefinition module)
-				?? MakeEmbeddedAttribute (module);
+			CompilerGeneratedAttributeTypeRef = FetchFromSystemRuntime (ModuleToEdit, "System.Runtime.CompilerServices", "CompilerGeneratedAttribute");
+			EmbeddedAttributeTypeDef = ModuleToEdit.Types.FirstOrDefault (td => td.FullName == "Microsoft.CodeAnalysis.EmbeddedAttribute")
+				?? MakeEmbeddedAttribute (ModuleToEdit);
 
-			EmbeddedAttributeTypeRef = module.ImportReference (new TypeReference (EmbeddedAttributeTypeDef.Namespace,
+			EmbeddedAttributeTypeRef = ModuleToEdit.ImportReference (new TypeReference (EmbeddedAttributeTypeDef.Namespace,
 				EmbeddedAttributeTypeDef.Name, EmbeddedAttributeTypeDef.Module, EmbeddedAttributeTypeDef.Scope));
 
 
-			NativeIntegerAttributeTypeDef = module.Types.FirstOrDefault (td => td.FullName == "System.Runtime.CompilerServices.NativeIntegerAttribute")
-				?? MakeNativeIntegerAttribute (module, EmbeddedAttributeTypeRef);
+			NativeIntegerAttributeTypeDef = ModuleToEdit.Types.FirstOrDefault (td => td.FullName == "System.Runtime.CompilerServices.NativeIntegerAttribute")
+				?? MakeNativeIntegerAttribute (ModuleToEdit, EmbeddedAttributeTypeRef);
 
 			NativeIntegerAttributeTypeRef = new TypeReference (NativeIntegerAttributeTypeDef.Namespace,
 				NativeIntegerAttributeTypeDef.Name, NativeIntegerAttributeTypeDef.Module, NativeIntegerAttributeTypeDef.Scope);
 
-			if (moduleMap.MicrosoftModule.AssemblyReferences.FirstOrDefault (an => an.Name == "System.Runtime.InteropServices") is AssemblyNameReference validReference) {
+			if (modules.MicrosoftModule.AssemblyReferences.FirstOrDefault (an => an.Name == "System.Runtime.InteropServices") is AssemblyNameReference validReference) {
 				InteropServicesAssembly = validReference;
 			} else {
-				throw new NotSupportedException ($"Assembly {moduleMap.MicrosoftModule.Name} does not have reference to System.Runtime.InteropServices. This is not possible.");
+				throw new NotSupportedException ($"Assembly {modules.MicrosoftModule.Name} does not have reference to System.Runtime.InteropServices. This is not possible.");
 			}
 
 			// if these type references aren't found in the module
 			// then we don't ever need to worry about reworking them
 			// and EmptyTypeReference is a fine substitute.
-			if (!module.TryGetTypeReference ("System.nint", out NintTypeReference)) {
+			if (!ModuleToEdit.TryGetTypeReference ("System.nint", out NintTypeReference)) {
 				NintTypeReference = EmptyTypeReference;
 			}
-			if (!module.TryGetTypeReference ("System.nuint", out NuintTypeReference)) {
+			if (!ModuleToEdit.TryGetTypeReference ("System.nuint", out NuintTypeReference)) {
 				NuintTypeReference = EmptyTypeReference;
 			}
-			if (!module.TryGetTypeReference ("System.nfloat", out NfloatTypeReference)) {
+			if (!ModuleToEdit.TryGetTypeReference ("System.nfloat", out NfloatTypeReference)) {
 				NfloatTypeReference = EmptyTypeReference;
+				NewNfloatTypeReference = EmptyTypeReference;
```

Comment:
Same, we could init all to be the empty ref and then just set them in the out param.